File: Test_Coverage/processed_dataset/tests/test_HumanEval_18.py
from functions.HumanEval_18 import how_many_times

def test_HumanEval_18():
     assert how_many_times('', 'x') == 0
     assert how_many_times('xyxyxyx', 'x') == 4
     assert how_many_times('cacacacac', 'cac') == 4
     assert how_many_times('john doe', 'john') == 1

File: Test_Coverage_Java/src/test/java/test_HumanEval_18.java
import org.junit.Test;
import static org.junit.Assert.assertEquals;

public class HumanEval18Test {

    @Test
    public void test_HumanEval_18_EmptyString() {
        assertEquals(0, howManyTimes("", "x"));
    }

    @Test
    public void test_HumanEval_18_SingleCharacterPattern() {
        assertEquals(4, howManyTimes("xyxyxyx", "x"));
    }

    @Test
    public void test_HumanEval_18_DuplicateCharactersInPattern() {
        assertEquals(4, howManyTimes("cacacacac", "cac"));
    }

    @Test
    public void test_HumanEval_18_SingleCharacterString() {
        assertEquals(1, howManyTimes("john doe", "john"));
    }
}